### ISSUE TITLE
Web API cached map features retrieval via Rect param

### ISSUE THREAD

**Issue by KOKAProduktion** [albar965/littlenavmap#790]
Hiho @albar965,

just when i thought I'm over the hill... :)

Adding clickable features to the map requires retrieving them by a `Rect` via the API. I've managed to do that already but I've stumbled upon the fact, that `MapQuery` applies caching (e.g. airportCache)...

Meaning that I'm able to retrieve features, but only those which are visible inside the in-app map instead of my provided `Rect` if I omit updating the cache. Doing so would make me feel uncomfortable, since it's the in-apps map cache...

POI:
https://github.com/KOKAProduktion/littlenavmap/blob/c89da645a63c8534ca7bef59ab4e8559c3243b09/src/query/mapquery.cpp#L734-L759

Also a `MapLayer` param is required for updating the cache and I'm afraid i don't understand the `MapLayer`s role in the setup as well as where to get the right one from.

Would you have some insights, hints or suggestions where to look further?
Should i implement a separate cache for the API map feature calls?

Thanks for your time and cheers!

**Comment by albar965** [comment]
Hope it helps. It is a tiny step for more than one map display in LNM. Something like an inset in the GUI might be possible.
I'm wondering why the old approach with the overlapping caches worked at all. :upside_down_face:

**Comment by KOKAProduktion** [comment]
Hi @albar965,

may i ask you again? I've successfully implemented correct airports retrieval via `Rect`. However VOR, NDB and Markers retrieval using the same pattern return "weird" results. It appears to me, that they misinterpret the provided `Rect` and return way too many or no results with coordinates in- and outside the original `Rect` depending on the distance value. It looks like the `Rect`s coordinates are somehow modulated by distance...

I must add that I'm converting from `atools::geo::Rect` to a `GeoDataLatLonBox` 
```
GeoDataLatLonBox latLonBox = GeoDataLatLonBox(rect.getNorth(),rect.getSouth(),rect.getEast(), rect.getWest());
```
So maybe I have introduced this issue myself... Although this works with the airports retrieval.

Did you encounter anything like this before?

Thanks and cheers!
